Sources/Client/Client/API/Client+Devices.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  Client+Devices.swift
+//  StreamChatCore
+//
+//  Created by Alexey Bukhtin on 09/01/2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+// MARK: - Devices
+
+public extension Client {
+    
+    /// Add a device for Push Notifications.
+    /// - Parameters:
+    ///   - deviceToken: a device token.
+    ///   - completion: an empty completion block.
+    @discardableResult
+    func addDevice(deviceToken: Data, _ completion: @escaping Client.Completion<EmptyData> = { _ in }) -> URLSessionTask {
+        if deviceToken.isEmpty {
+            completion(.failure(.emptyDeviceToken))
+            return .empty
+        }
+        
+        return addDevice(deviceId: deviceToken.deviceToken, completion)
+    }
+    
+    /// Add a device for Push Notifications.
+    /// - Parameters:
+    ///   - deviceId: a Push Notifications device identifier.
+    ///   - completion: an empty completion block.
+    @discardableResult
+    func addDevice(deviceId: String, _ completion: @escaping Client.Completion<EmptyData> = { _ in }) -> URLSessionTask {
+        guard let currentUser = user else {
+            completion(.failure(.emptyUser))
+            return .empty
+        }
+        
+        let device = Device(deviceId)
+        
+        if currentUser.devices.contains(where: { $0.id == deviceId }) {
+            if currentUser.currentDevice == nil {
+                var currentUser = currentUser
+                currentUser.currentDevice = device
+                user = currentUser
+            }
+            
+            completion(.success(.empty))
+            return .empty
+        }
+        
+        let completion = beforeCompletion(completion) { [unowned self] _ in
+            if let currentUser = self.user {
+                var currentUser = currentUser
+                currentUser.devices.append(device)
+                currentUser.currentDevice = device
+                self.user = currentUser
+                self.logger?.log("📱 Device added with id: \(deviceId)")
```

Comment:
we should stop using non ascii chars in out logs; this is just asking for beardy dev-ops to hate us

Reply:
It's really useful. It's local logs.

Reply:
Have we received any complains yet?